### PR TITLE
update guide to reflect tenancy and scope

### PR DESCRIPTION
**Pull request by dhiaayachi**
### Description

This PR update the resource dev guide to reflect the changes introduced by supporting tenancy v1 when using v2 resource API

### Testing & Reproduction steps

Doc change, not test needed

### CODE REVIEW

File: docs/resources/guide.md [hunk 1]
```diff
@@ -172,7 +178,8 @@ a set of ACL hooks:
 func RegisterTypes(r resource.Registry) {
 	r.Register(resource.Registration{
 		Type:  BarV1Alpha1Type,
-		Proto: &pbv1alpha1.Bar{},
+		Proto: &pbv1alpha1.Bar{}, 
+		Scope:    resource.ScopeNamespace,
```
Comment (analogue) [inline docs/resources/guide.md:182]:
```suggestion
		Proto: &pbv1alpha1.Bar{}, 
		Scope: resource.ScopeNamespace,
```

**Comment by analogue** [comment]
Can you update L191-L204 to match the current ACL hooks? 

```go
type ACLHooks struct {
	// Read is used to authorize Read RPCs and to filter results in List
	// RPCs.
	//
	// If it is omitted, `operator:read` permission is assumed.
	Read func(acl.Authorizer, *acl.AuthorizerContext, *pbresource.ID) error

	// Write is used to authorize Write and Delete RPCs.
	//
	// If it is omitted, `operator:write` permission is assumed.
	Write func(acl.Authorizer, *acl.AuthorizerContext, *pbresource.Resource) error

	// List is used to authorize List RPCs.
	//
	// If it is omitted, we only filter the results using Read.
	List func(acl.Authorizer, *acl.AuthorizerContext) error
}
```

**Review by analogue — CHANGES_REQUESTED** [review]
Just minor formatting updates.

File: docs/resources/guide.md [hunk 2]
```diff
@@ -50,16 +50,21 @@ import (
 var BarV1Alpha1Type = &pbresource.Type{
 	Group:        "foo",
 	GroupVersion: "v1alpha1",
+	Scope:    resource.ScopeNamespace,
 	Kind:         "bar",
 }
```
[the remark this answers is not in the document]
Comment (dhiaayachi) [reply docs/resources/guide.md:55]: @analogue is removing the scope here on purpose?

[the remark this answers is not in the document]
Comment (analogue) [reply docs/resources/guide.md:55]: Yes, `Scope` isn't a field in `pbresource.Type`.

[the remark this answers is not in the document]
Comment (dhiaayachi) [reply docs/resources/guide.md:55]: 🤦
my bad!

**Review by analogue — APPROVED** [review]
🚀

**Pull request merged**